fbw-a32nx/src/systems/instruments/src/PFD/AttitudeIndicatorFixed.tsx

```diff
@@ -12,515 +12,602 @@ import { Arinc429Values } from './shared/ArincValueProvider';
+    const foStickDisabledFcdc2 = this.fcdc2DiscreteWord2.getBitValueOr(29, false);
+    const captStickDisabledFcdc1 = this.fcdc1DiscreteWord2.getBitValueOr(28, false);
+    const captStickDisabledFcdc2 = this.fcdc2DiscreteWord2.getBitValueOr(28, false);
+    const foStickDisabled = foStickDisabledFcdc1 || foStickDisabledFcdc2;
+    const captStickDisabled = captStickDisabledFcdc1 || captStickDisabledFcdc2;
 
-        if (!showIndicator) {
-            this.onGroundForVisibility.set('hidden');
-        } else {
-            this.onGroundForVisibility.set('visible');
-            this.crossHairRef.instance.style.transform = `translate3d(${totalRollCommand}px, ${totalPitchCommand}px, 0px)`;
-        }
-    }
-
-    onAfterRender(node: VNode): void {
-        super.onAfterRender(node);
-
-        const sub = this.props.bus.getSubscriber<PFDSimvars & Arinc429Values>();
-
-        sub.on('noseGearCompressed').whenChanged().handle((g) => {
-            this.onGround = g;
-            this.handleSideStickIndication();
-        });
-
-        sub.on('engOneRunning').whenChanged().handle((e) => {
-            this.engOneRunning = e;
-            this.handleSideStickIndication();
-        });
-
-        sub.on('engTwoRunning').whenChanged().handle((e) => {
-            this.engTwoRunning = e;
-            this.handleSideStickIndication();
-        });
-
-        sub.on('fcdc1DiscreteWord2').whenChanged().handle((discreteWord2) => {
-            this.fcdc1DiscreteWord2 = discreteWord2;
-            this.handleSideStickIndication();
-        });
-
-        sub.on('fcdc2DiscreteWord2').whenChanged().handle((discreteWord2) => {
-            this.fcdc2DiscreteWord2 = discreteWord2;
-            this.handleSideStickIndication();
-        });
-
-        sub.on('fcdcCaptPitchCommand').whenChanged().handle((x) => {
-            this.captPitchCommand = x;
-            this.handleSideStickIndication();
-        });
-
-        sub.on('fcdcFoPitchCommand').whenChanged().handle((x) => {
-            this.foPitchCommand = x;
-            this.handleSideStickIndication();
-        });
-
-        sub.on('fcdcCaptRollCommand').whenChanged().handle((y) => {
-            this.captRollCommand = y;
-            this.handleSideStickIndication();
-        });
-
-        sub.on('fcdcFoRollCommand').whenChanged().handle((y) => {
-            this.foRollCommand = y;
-            this.handleSideStickIndication();
-        });
-    }
+    const totalPitchCommand =
+      Math.max(
+        Math.min(
+          (foStickDisabled ? 0 : this.foPitchCommand.value) + (captStickDisabled ? 0 : this.captPitchCommand.value),
+          16,
+        ),
+        -16,
+      ) * -1.43875;
+    const totalRollCommand =
+      Math.max(
+        Math.min(
+          (foStickDisabled ? 0 : this.foRollCommand.value) + (captStickDisabled ? 0 : this.captRollCommand.value),
+          20,
+        ),
+        -20,
+      ) * 1.478;
```

Comment:
Expanded like this it is a bit hard to follow. Maybe more because of the magic numbers.  Potentially add a TODO here.